--- src/propgrid.cpp	2009-10-12 20:59:03.000000000 -0700
+++ propgrid.cpp	2009-10-12 20:57:55.000000000 -0700
@@ -3450,7 +3450,7 @@ void wxPropertyGrid::Thaw()
     // Force property re-selection.
     // NB: We must copy the selection.
     wxArrayPGProperty selection = m_pState->m_selection;
-    DoSetSelection(selection, wxPG_SEL_FORCE);
+    DoSetSelection(selection, wxPG_SEL_FORCE|wxPG_SEL_NONVISIBLE);
 }
 
 // -----------------------------------------------------------------------
